src/Sarif.Viewer.VisualStudio/ProjectNameLookup.cs

```diff
@@ -0,0 +1,33 @@
+﻿using System.Collections.Generic;
+
+namespace Microsoft.Sarif.Viewer
+{
+    public sealed class ProjectNameLookup
+    {
+        private readonly static ProjectNameLookup _instance = new ProjectNameLookup();
+        private Dictionary<string, string> _projectNames = new Dictionary<string, string>();
+
+        static ProjectNameLookup() { }
+        private ProjectNameLookup() { }
+
+        public static ProjectNameLookup Instance => _instance;
```

Comment:
>Instance [](start = 40, length = 8)

Do we need to worry about thread safety? Do we have to do this?

```
private readonly thisLock = new object();
...
public static ProjectNameLookup Instance
{
    get
    {
        lock(thisLock)
        {
            return instance;
        }
    }
}
```

Reply:
What happens if two threads try to enter SetName around the same time?  Would it run the FindProjectItem method numerous times for the same fileName?  If so, would that be cheaper performance wise than taking out a lock every time we call the method?  What are your thoughts?

---
In reply to: [135165342](https://github.com/Microsoft/sarif-sdk/pull/596#discussion_r135165342) [](ancestors = 135165342)